projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -768,7 +829,7 @@ export class IgxDateRangePickerComponent extends DisplayDensityBase
     private extractRange(selection: Date[]): DateRange {
         return {
             start: selection[0],
-            end: selection.length > 1 ? selection[selection.length - 1] : null
+            end: selection.length > 0 ? selection[selection.length - 1] : null
```

Comment:
Okay, so that ensures with one selection you get a range of the same start date? Does it affect anything else in terms of behavior?

Reply:
`extractRange` is called only in `handleSelection`. Last one is called at two places:
- in calendar as handler for `onSelection`.
- in public `selectRange` method. This one says _"Selects a range of dates. If no `endDate` is passed, range is 1 day (only `startDate`)"_
I think the fix is ok.